src/mumble/JackAudio.cpp

```diff
@@ -1059,6 +1094,77 @@ bool JackAudioOutput::process(const jack_nframes_t frames) {
 	return true;
 }
 
+void JackAudioOutput::prepareOutputBuffers(unsigned int frameCount, QList< AudioOutputBuffer * > *qlMix,
+										   QList< AudioOutputBuffer * > *qlDel) {
+	ServerHandlerPtr sh = Global::get().sh;
+	VoiceRecorderPtr recorder;
+	if (sh) {
+		recorder = Global::get().sh->recorder;
+	}
+
+	// Keep track of ports we've filled the audio of.
+	QList< jack_port_t * > qlPortsYetToFill(userPorts.values());
+
+	// Register user ports based on who is currently present in qmOutputs and route their audio to JACK.
+	// Don't care if users get removed for now. TODO: maybe at some point we do.
+	QMultiHash< const ClientUser *, AudioOutputBuffer * >::const_iterator it = qmOutputs.constBegin();
+	while (it != qmOutputs.constEnd()) {
+		const ClientUser *user   = it.key();
+		AudioOutputBuffer *audio = it.value();
+
+		if (!user || !recorder || (recorder && !recorder->isTransportEnabled())) {
+			if (audio->prepareSampleBuffer(frameCount)) {
+				qlMix->append(audio);
+			} else {
+				qlDel->append(audio);
+			}
+
+			++it;
+			continue;
+		}
+
+		QString qsPortName = user->qsName;
+		qsPortName         = qsPortName.prepend("user_");
+		if (!userPorts[qsPortName]) {
+			auto port = jas->registerPort(qsPortName.toStdString().c_str(), JackPortIsOutput);
+			if (!port) {
+				qWarning("JackAudioOutput: unable to register user port \"%s\"", qsPortName.toStdString().c_str());
+			} else {
+				userPorts[qsPortName] = port;
+			}
+		}
+
+		if (audio->prepareSampleBuffer(frameCount)) {
+			qlPortsYetToFill.removeOne(userPorts[qsPortName]);
+			qlMix->append(audio);
+
+			auto outputBuffer = (float *) jas->getPortBuffer(userPorts[qsPortName], frameCount);
```

Comment:
Don't use C-style casts. Instead, use `static_cast` or if necessary `reinterpret_cast `

Reply:
I wondered about this, the main lang I write in is C# and I didn't know what the difference between a C-style, `static_cast` or `reinterpret_cast` was. I'll research and choose the suitable substitute

Reply:
> I'll research and choose the suitable substitute

First try static_cast and if the compiler yells at you, use the other :D

Reply:
Thanks for the tip! 😄